core/actor/src/main/scala/net/liftweb/actor/LAFuture.scala

```diff
@@ -295,13 +300,13 @@ object LAFuture {
     }
   }
 
-  private def executeWithObservers(scheduler: LAScheduler, f: () => Unit) {
+  private def executeWithObservers(scheduler: LAScheduler, f: () => Unit, wrapper:CommonLoanWrapper) {
     val cur = threadInfo.get()
     scheduler.execute(() => {
       val old = threadInfo.get()
       threadInfo.set(cur)
       try {
-        f()
+        wrapper(f())
```

Comment:
Right, I fault on the side of the code being clear more than guessing at how it may perform when it comes to something as sophisticated as the JVM.  I can really go either way here.  Do you still think the Option is a better route?